Framework/Algorithms/test/PolarizationEfficiencyCorTest.h

```diff
@@ -50,363 +50,253 @@ class PolarizationEfficiencyCorTest : public CxxTest::TestSuite {
-    PolarizationEfficiencyCor alg;
-    alg.setRethrows(true);
-    alg.initialize();
-    alg.setProperty("OutputWorkspace", "out");
-    alg.setProperty("InputWorkspaceGroup", createWorkspaceGroup(4));
-    alg.setProperty("CorrectionMethod", "Fredrikze");
-    alg.setProperty("Efficiencies", createEfficiencies("Fredrikze"));
-    alg.setProperty("Flippers", "00, 01, 10, 11");
+    auto alg = createAlgorithm(FREDRIKZE_METHOD, FREDRIKZE_METHOD);
+    alg->setProperty("InputWorkspaceGroup", createWorkspaceGroup(4));
+    alg->setProperty("Flippers", "00, 01, 10, 11");
     // Error: Property Flippers canot be used with the Fredrikze method
-    TS_ASSERT_THROWS(alg.execute(), const std::invalid_argument &);
+    TS_ASSERT_THROWS_EQUALS(alg->execute(), const std::invalid_argument &e, std::string(e.what()),
+                            "Property Flippers cannot be used with the Fredrikze method.");
+  }
+
+  void test_that_wildes_output_spinstates_cannot_be_used_with_fredrikze() {
+    auto alg = createAlgorithm(FREDRIKZE_METHOD, FREDRIKZE_METHOD);
+    alg->setProperty("InputWorkspaceGroup", createWorkspaceGroup(4));
+    alg->setPropertyValue("SpinStatesOutWildes", "+, -");
+
+    TS_ASSERT_THROWS_EQUALS(alg->execute(), const std::invalid_argument &e, std::string(e.what()),
+                            "Property SpinStatesOutWildes cannot be used with the Fredrikze method.");
   }
 
   void test_histo() {
-    PolarizationEfficiencyCor alg;
-    alg.setRethrows(true);
-    alg.initialize();
-    alg.setProperty("OutputWorkspace", "out");
-    alg.setProperty("InputWorkspaces", createWorkspacesInADS(4));
-    alg.setProperty("CorrectionMethod", "Wildes");
-    alg.setProperty("Efficiencies", createEfficiencies("histo"));
-    alg.execute();
-    WorkspaceGroup_sptr out = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>("out");
-    TS_ASSERT_EQUALS(out->size(), 4);
+    auto alg = createAlgorithm("histo", WILDES_METHOD);
+    alg->setProperty("InputWorkspaces", createWorkspacesInADS(4));
+    alg->execute();
+    checkOutputWorkspaceGroupSize(4);
   }
 
   void test_points() {
-    PolarizationEfficiencyCor alg;
+    auto alg = createAlgorithm("points", WILDES_METHOD);
     auto const inputs = createWorkspacesInADS(4);
-    alg.setRethrows(true);
-    alg.initialize();
-    alg.setProperty("OutputWorkspace", "out");
-    alg.setProperty("InputWorkspaces", inputs);
-    alg.setProperty("CorrectionMethod", "Wildes");
-    alg.setProperty("Efficiencies", createEfficiencies("points"));
-    alg.execute();
-    WorkspaceGroup_sptr out = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>("out");
-    TS_ASSERT_EQUALS(out->size(), 4);
+    alg->setProperty("InputWorkspaces", inputs);
+    alg->execute();
+    WorkspaceGroup_sptr out = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(OUTPUT_GRP_NAME);
+    checkOutputWorkspaceGroupSize(4);
 
     for (size_t i = 0; i < out->size(); ++i) {
```

Comment:
In `test_points` and `test_points_short` we're comparing the output and input workspaces using the `CompareWorkspaces` algorithm. As this code for setting up and calling this algorithm is duplicated between the two tests, could we pull this into a helper method?